packages/repository/src/repositories/relation.repository.ts

```diff
@@ -26,7 +30,47 @@ export interface HasManyEntityCrudRepository<T extends Entity, ID> {
    * @param options Options for the operation
    * @returns A promise which resolves with the found target instance(s)
    */
-  find(filter?: Filter | undefined, options?: Options): Promise<T[]>;
+  find(filter?: Filter, options?: Options): Promise<T[]>;
+  /**
+   * 
+   * @param entity 
+   * @param options 
+   */
+  update(entity: DataObject<T>, options?: Options): Promise<boolean>;
+  /**
+   * 
+   * @param entity 
+   * @param options 
+   */
+  delete(entity: DataObject<T>, options?: Options): Promise<boolean>;
+  /**
+   * 
+   * @param entity 
+   * @param options 
+   */
+  save(entity: DataObject<T>, options?: Options): Promise<T | null>;
+  /**
+   * 
+   * @param dataObjects 
+   */
+  createAll(dataObjects: DataObject<T>[]): Promise<T[]>;
+  /**
+   * 
+   * @param where 
+   * @param options 
+   */
+  deleteAll(where?: Where, options?: Options): Promise<number>;
+  /**
+   * 
+   * @param dataObject 
+   * @param where 
+   * @param options 
+   */
+  updateAll(
```

Comment:
Is this going to perform a partial update (patch) or a full replace? I believe our connectors are performing a partial update, let's call this method `patchAll`.

I am also not sure if the suffix `All` is descriptive enough, can we come up with a better name please? E.g. `patchMany` or `patchByQuery`.